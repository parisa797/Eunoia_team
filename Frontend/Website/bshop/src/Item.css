
.item-page{
    padding: 5vh 5vw;
    direction: rtl;
    font-family: Vazir;
    color: var(--font-color2);
}


.page-content .my-3{
    background-color: var(--primary-color);
}


.item-page .description
{
    margin: auto;
    display: flex;
    padding: 5vh 5vw;
    font-size: 18px;
}

.item-title{
    color: var(--primary-color);
}


.item-page .btn{
    background-color: var(--primary-color);
    border: 1px solid black;
    color: white;
}

.item-page .w3-hover-sepia {
        margin-top:200px;
        top: 200vw;
        width: 250px;
        height: 250px;
    }

    .column {
        margin: auto;
        display: flex;
        flex-direction: column;

    }


    .sub-container {
        display: flex;
        width: 100%;
    }

.item-page .item-title
{
    margin: auto;
}
.item-page .imageWrapper{
    position: relative;
}
.item-page .image{
    width: 100%;
    height: auto;
    position: absolute;
    right: 0;
    top: 50%;
    transform: translate(0, -50%);
}
.item-page .stars{
    overflow: initial !important;
}

.item-page .item-comment-container{
    width: 100%;
    padding: 0%;
}

.item-page .count-div{
    display: inline-flex;
    /* grid-template-columns: 2rem auto 2rem; */
    height: fit-content;
    width:100%;
    padding: 0 5px;
    margin: 5px auto;
}

.item-page .count-div .btn{
    color: var(--primary-color);
    font-size: x-large;
    padding-top: 0;
    padding-bottom: 0;
    border: none;
    background: transparent;
}

.item-page .cart-btn:active{
    color: var(--primary-color);
    background: transparent;
}

.item-page .count-div input{
    width: 5rem;
    overflow: hidden;
    text-align: center;
}