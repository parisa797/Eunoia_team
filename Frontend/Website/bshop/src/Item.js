import React, { useState, useEffect } from "react";
import "./Item.css";
import ReactStars from "react-rating-stars-component";
import ShopSideBar from "./ShopSideBar";
import Itemcomment from "./Itemcomment";
import FavoriteIcon from '@material-ui/icons/Favorite';
import FavoriteBorderIcon from '@material-ui/icons/FavoriteBorder';
import { IconButton } from "@material-ui/core";
function Item(props) {
  // console.log("Token "+localStorage.getItem("token"))
  const [items, setItems] = useState({});
  const [rated, setRated] = useState(false)
  const [rateID, setRateId] = useState(null)
  const[triggerReload,setTriggerReload] = useState(false)
  var shopID = window.location.pathname.match(/[^\/]+/g)[1];
  var itemID = window.location.pathname.match(/[^\/]+/g)[3];
  // let itemId = window.location.pathname.match.items(/[^\/]+/g)[3]
  // let shopID = window.location.pathname.match(/[^\/]+/g)[1]
  useEffect(() => {
    fetch("http://eunoia-bshop.ir:8000/shops/" + shopID + "/items/" + itemID, {
      method: "GET",
      // headers: {
      //     "Authorization": "Token " + localStorage.getItem('token')
      // }
    })
      .then((res) => {
        if (res.status === 200) {
          return res.json();
        }
      })
      .then((res) => {
        setItems(res);
      });
  }, [triggerReload]);
  useEffect(() => {
    console.log(props.userState)
    fetch("http://eunoia-bshop.ir:8000/shops/"+shopID+"/items/"+itemID+"/rates/" , {
        method: 'GET',
        headers: {
            "Authorization": "Token " + localStorage.getItem('token')
        }
    })
        .then((res) => {
            if (res.status === 200) {
                return res.json()
            }
        }
        ).then((res) => {
            console.log("in list rate")
            console.log(res)
            let username = localStorage.getItem("username")
            for (let i in res) {
                if (!!res[i].user && res[i].user.user_name === username) {
                    setRated(true)
                    setRateId(res[i].id)
                }
            }
        })
    // })
}, [])

  useEffect(() => {}, []);
  function ratingChanged(new_rating) {
    let fd = new FormData()
    console.log(rated)
    fd.append("rate", new_rating)
    if (!rated) {
        fd.append("item", itemID)
        fetch("http://eunoia-bshop.ir:8000/shops/"+shopID+"/items/"+itemID+"/rates/", {
            method: 'POST',
            headers: {
                "Authorization": "Token " + localStorage.getItem('token')
            }
            ,
            body: fd
        }).then(res => {
            if (res.status === 201) {

                return res.json()
            }
            console.log(res.status)
            return null
        }).then(res => {
            if (res) {
                setRateId(res.id)
                setRated(true)
                setTriggerReload(!triggerReload)
            }

        })
    }
    else {
        fetch("http://eunoia-bshop.ir:8000/shops/"+shopID+"/items/"+itemID+"/rates/" + rateID, {
            method: 'PUT',
            headers: {
                "Authorization": "Token " + localStorage.getItem('token')
            }
            ,
            body: fd
        }).then(res => {
            if (res.status === 200) {
                setRated(true)
                setTriggerReload(!triggerReload)
            }
            console.log(res.status)
        })
    }

}

  console.log(items);
  const [isLikedByUser, setisLikedByUser] = useState(false)
  useEffect(() => {
      fetch(`http://eunoia-bshop.ir:8000/shops/${shopID}/items/${itemID}/likes`,{
          method: 'GET',
          headers: {
              "Authorization": "Token " + localStorage.getItem('token')
          }
      }).then(res => res.json())
      .then(res => {
        const username = localStorage.getItem("username")
        const resp = res[0]?.Liked_By.findIndex(i => i.username === username)
        if(resp > -1) setisLikedByUser(true)
        else setisLikedByUser(false)
      })
  }, [])
  const handleLikeItem = async () => {
      await fetch(`http://eunoia-bshop.ir:8000/shops/${shopID}/items/${itemID}/likes`,{
          method: 'POST',
          headers: {
              "Authorization": "Token " + localStorage.getItem('token')
          }
      })
      fetch(`http://eunoia-bshop.ir:8000/shops/${shopID}/items/${itemID}/likes`,{
        method: 'GET',
        headers: {
            "Authorization": "Token " + localStorage.getItem('token')
        }
    }).then(res => res.json())
    .then(res => {
      const username = localStorage.getItem("username")
      const resp = res[0]?.Liked_By.findIndex(i => i.username === username)
      if(resp > -1) setisLikedByUser(true)
      else setisLikedByUser(false)
    })
  }
  // const handleLikeItem = () => {

  // }
  const checkIsLikedByUserOrNot = () => {
    return true
  }
  return (
    <div style={{ padding: "5vh 2vw" }}>
      <ShopSideBar />
      <div className="page-contents" data-testid="item">
        <div className="item-page">
          <div className="sub-container">
            <div className="col-12 d-flex flex-wrap">
              <div className="col-sm-12 col-md-5 imageWrapper">
                <img className=" image " src={items.photo} alt="" />
              </div>
              <div className="lead text-right col-sm-12 col-md-6">
                <h1 className="my-3" data-testid="item-name">{items.name}</h1>
                <p className="shop-comment-date">
                    <IconButton onClick={() => handleLikeItem()} style={{ color: 'red', padding: '0' }}> 
                        {isLikedByUser ? <FavoriteIcon /> : <FavoriteBorderIcon />}
                    </IconButton>

                </p>
                <hr></hr>
                {!!items.description && (
                  <>
                    <h3 className="text-right mb-3">ویژگی های کالا:</h3>
                    <p className="lead text-right" data-testid="item-description">{items.description}</p>
                  </>
                )}
                <div className="rating-item">
                            <> <p style={{ direction: "ltr" }} data-testid={"item-rate-value"}>امتیاز: {items.rate_value?Math.round(items.rate_value * 10) / 10 :0} </p>
                                    {!!items.rate_value && <ReactStars
                                        edit={props.userState!=="u"}
                                        value={Math.round(items.rate_value)}
                                        isHalf={false}
                                        classNames="stars"
                                        data-testid="item-rate-stars"
                                        size={25}
                                        onChange={ratingChanged}
                                        activeColor={"var(--primary-color)"}
                                    />}
                                    {!items.rate_value && <ReactStars
                                        edit={!!localStorage.getItem("token")}
                                        value={0}
                                        isHalf={false}
                                        classNames="stars"
                                        data-testid="shop-rate-stars"
                                        size={25}
                                        onChange={ratingChanged}
                                        activeColor={"var(--primary-color)"}
                                    /> }
                                    <p data-testid={"item-rate-count"}>‌({items.rate_count})</p>
                                </>
                            </div>
                 {/* <div className="items-stars">
                                    <ReactStars
                                        edit={false}
                                        value={items.rate_value ? items.rate_value : 0}
                                        isHalf={true}
                                        classNames="stars"
                                        size={20}
                                        activeColor={"var(--primary-color)"}
                                    />
                                    <p data-testid={"region-items-rate-count" + items.id}>({items.rate_count})</p>
                                </div> */}
                {!!items.count && (
                  <div className="col-12 d-flex flex-justify-between p-0 my-2">
                    <div className="col-6 p-0 text-right">موجودی:</div>
                    <div className="col-6 p-0 ">{items.count}</div>
                  </div>
                )}
                {!!items.price && (
                  <div className="col-12 d-flex flex-justify-between p-0 my-2">
                    <div className="col-6 p-0 text-right">قیمت:</div>
                    <div className="col-6 p-0 ">{items.price} ریال</div>
                  </div>
                )}
                {!!items.manufacture_jalali && (
                  <div className="col-12 d-flex flex-justify-between p-0 my-2">
                    <div className="col-6 p-0 text-right">تاریخ تولید:</div>
                    <div className="col-6 p-0 " data-testid="item-manufacture_jalali">{items.manufacture_jalali}</div>
                  </div>
                )}
                 {!!items.Expiration_jalali && (
                  <div className="col-12 d-flex flex-justify-between p-0 my-2">
                    <div className="col-6 p-0 text-right">تاریخ انقضا:</div>
                    <div className="col-6 p-0 " data-testid="item-Expiration_jalali">{items.Expiration_jalali}</div>
                  </div>
                )}

              {props.userState !== "m" && <a href="#" className="btn btn-lg btn-primary my-3" >افزودن به سبد خرید</a>}
              </div>
              <div className="col-12 item-comment-container" >
                <Itemcomment userState={props.userState} />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
}
export default Item;

{
  /* {items.ItemShop && items.ItemShop.phone} */
}