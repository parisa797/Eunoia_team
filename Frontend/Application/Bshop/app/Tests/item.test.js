import React from "react";
import {
  render,
  fireEvent,
  cleanup,
  waitFor,
  act,
} from "react-native-testing-library";
import renderer from "react-test-renderer";

import ShopDetail from "../shopDetails";
import Item from "../item";

const mockedFocused = jest.fn();
jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useIsFocused: () => mockedFocused,
}));

describe("testing shop detail, items part", () => {
  it("renders correctly", () => {
    // const params = jest.fn();
    const params = {
      id: 3,
      logo: "hi.png",
      title: "myshop",
      address: "myStreet",
      phone: "09123456789",
      online: true,
      rate_value: 3,
    };
    const tree = renderer.create(<ShopDetail route={{ params }} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("fetch test", async () => {
    const fetchMock = require("fetch-mock-jest");
    const shopitems = [
      {
        name: "پف پفی شیبابا",
        price: 80000,
        photo: "/pofpofi.png",
        id: 0,
        discount: 0,
      },
      {
        name: "پفک طلایی چی توز",
        price: 100000,
        photo: "pofak.png",
        id: 1,
        discount: 10,
      },
      {
        name: "پاستیل شیبابا",
        price: 50000,
        photo: "pastil.jpg",
        id: 2,
        discount: 20,
      },
      {
        name: "رب گوجه آتا",
        price: 300000,
        photo: "rob.png",
        id: 3,
        discount: 0,
      },
    ];

    var params = {
      id: 3,
      logo: "hi.png",
      title: "myshop",
      address: "myStreet",
      phone: "09123456789",
      online: true,
      rate_value: 3,
    };
    var url = "http://eunoia-bshop.ir:8000/shops/" + params.id + "/items/";
    fetchMock.get(url, shopitems);

    const { getByTestId, queryByTestId } = await render(
      <ShopDetail route={{ params }} />
    );
    await new Promise((resolve) => setImmediate(resolve));

    expect(fetchMock).toHaveFetched(
      "http://eunoia-bshop.ir:8000/shops/3/items/",
      "get"
    );
    expect(getByTestId("items-list").props.data.length).toBe(shopitems.length);

    for (var i = 0; i < shopitems.length; i++) {
      expect(getByTestId("item-name-" + i).props.children).toBe(
        shopitems[i].name
      );

      var expected_image = "http://eunoia-bshop.ir:8000" + shopitems[i].photo;
      expect(getByTestId("item-image-" + i).props.source.uri).toBe(
        expected_image
      );

      // if item has some discount
      if (shopitems[i].discount != 0) {
        console.log("disc wasn't zero");
        var price = ["قیمت:", shopitems[i].price];
        var newPrice = [
          "قیمت با تخفیف: ",
          // (
          ((100 - shopitems[i].discount) * shopitems[i].price) / 100,
          // ).toString(),
        ];
        expect(queryByTestId("item-price1-" + i)).toBeNull();
        expect(queryByTestId("item-price0-" + i).props.children).toStrictEqual(
          price
        );
        expect(
          queryByTestId("item-discounted-" + i).props.children
        ).toStrictEqual(newPrice);
      } else {
        var price = ["قیمت:", shopitems[i].price];

        expect(queryByTestId("item-price0-" + i)).toBeNull();
        expect(queryByTestId("item-price1-" + i).props.children).toStrictEqual(
          price
        );
        expect(queryByTestId("item-discounted-" + i)).toBeNull();
      }
    }

    fetchMock.mockClear();
  });
});
