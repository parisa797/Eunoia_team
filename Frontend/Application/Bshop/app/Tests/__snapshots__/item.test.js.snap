// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing shop detail, items part renders correctly 1`] = `
<RCTScrollView
  nestedScrollEnabled={true}
  style={
    Object {
      "flex": 1,
      "marginBottom": "5%",
      "marginTop": "1%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "backgroundColor": "#b31414",
          "borderRadius": 10,
          "elevation": 5,
          "fontSize": 25,
          "fontWeight": "bold",
          "height": 180,
          "marginLeft": 15,
          "marginTop": 5,
          "shadowColor": "black",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.26,
          "width": 370,
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "height": 50,
            "marginBottom": "5%",
            "marginLeft": 80,
            "marginTop": "1%",
            "width": "76%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="جستجو"
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Object {
              "flex": 1,
              "fontSize": 20,
              "height": 50,
              "padding": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 20,
            "marginLeft": 6,
            "marginTop": -68,
            "opacity": 1,
            "width": "20%",
          }
        }
      >
        <Text>
          بگرد
        </Text>
      </View>
      <View
        style={
          Object {
            "color": "white",
            "marginLeft": 10,
            "marginTop": -10,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
            }
          }
        >
          فیلتر آیتم بر اساس 
        </Text>
        <View>
          <View
            style={
              Object {
                "color": "white",
                "height": 50,
                "width": 150,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "همه",
                    "textColor": undefined,
                    "value": "all",
                  },
                  Object {
                    "label": "ادویه، چاشنی و مخلفات غذا",
                    "textColor": undefined,
                    "value": "Spices and condiments and food side dishes",
                  },
                  Object {
                    "label": "بهداشت و مراقبت پوست",
                    "textColor": undefined,
                    "value": "Cosmetics",
                  },
                  Object {
                    "label": "آرایش و پیرایش",
                    "textColor": undefined,
                    "value": "Makeup and trimming",
                  },
                  Object {
                    "label": "پروتئینی",
                    "textColor": undefined,
                    "value": "Protein",
                  },
                  Object {
                    "label": "تنقلات",
                    "textColor": undefined,
                    "value": "Junk Food",
                  },
                  Object {
                    "label": "خشکبار",
                    "textColor": undefined,
                    "value": "Nuts",
                  },
                  Object {
                    "label": "شیرینیجات و دسرها",
                    "textColor": undefined,
                    "value": "Sweets and desserts",
                  },
                  Object {
                    "label": "عطر، ادکلن و اسپری",
                    "textColor": undefined,
                    "value": "perfume",
                  },
                  Object {
                    "label": "غذا، کنسرو و سبزیجات",
                    "textColor": undefined,
                    "value": "Fruits and vegetables",
                  },
                  Object {
                    "label": "لبنیات",
                    "textColor": undefined,
                    "value": "Dairy",
                  },
                  Object {
                    "label": "نوشیدنیها",
                    "textColor": undefined,
                    "value": "Drinks",
                  },
                  Object {
                    "label": "وسایل شستشو و نظافت",
                    "textColor": undefined,
                    "value": "Washing and Cleaning Equipment",
                  },
                  Object {
                    "label": "متفرقه",
                    "textColor": undefined,
                    "value": "others",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
        <View>
          <View
            style={
              Object {
                "color": "white",
                "height": 50,
                "width": 150,
              }
            }
          >
            <RCTPicker
              items={
                Array [
                  Object {
                    "label": "گران ترین",
                    "textColor": undefined,
                    "value": "expensive",
                  },
                  Object {
                    "label": "ارزان ترین",
                    "textColor": undefined,
                    "value": "cheap",
                  },
                  Object {
                    "label": "جدیدترین",
                    "textColor": undefined,
                    "value": "new",
                  },
                  Object {
                    "label": "دسته بندی",
                    "textColor": undefined,
                    "value": "category",
                  },
                  Object {
                    "label": "تخفیف دار",
                    "textColor": undefined,
                    "value": "discount",
                  },
                ]
              }
              onChange={[Function]}
              selectedIndex={0}
              style={
                Array [
                  Object {
                    "height": 216,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "height": 30,
            "justifyContent": "center",
            "marginBottom": 20,
            "marginLeft": 6,
            "marginTop": -120,
            "opacity": 1,
            "width": "20%",
          }
        }
      >
        <Text>
          فیلتر
        </Text>
      </View>
    </View>
    <View>
      <View
        style={
          Object {
            "backgroundColor": "#f1f1f2",
            "borderRadius": 10,
            "elevation": 5,
            "height": 320,
            "marginLeft": 18,
            "marginTop": 10,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.26,
            "width": 370,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "elevation": 5,
              "height": "40%",
              "marginLeft": 110,
              "marginTop": 10,
              "shadowColor": "black",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.26,
              "width": "39%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "hi.png",
              }
            }
            style={
              Object {
                "alignItems": "center",
                "height": "80%",
                "justifyContent": "center",
                "marginLeft": 12,
                "marginTop": "12%",
                "resizeMode": "cover",
                "width": "80%",
              }
            }
          />
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#b31414",
              "borderRadius": 100,
              "height": 40,
              "justifyContent": "center",
              "marginBottom": 20,
              "marginLeft": 10,
              "marginTop": -60,
              "opacity": 1,
              "width": 70,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
             نظرات
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "height": "38%",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 20,
                "marginLeft": "-1%",
                "marginTop": "1%",
                "marginVertical": 5,
              }
            }
          >
            فروشگاه 
            myshop
          </Text>
          <View
            style={
              Object {
                "alignItems": "baseline",
                "flexDirection": "row-reverse",
                "justifyContent": "space-between",
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                }
              }
            >
              myStreet
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "baseline",
                "flexDirection": "row-reverse",
                "justifyContent": "space-between",
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "black",
                  "fontSize": 20,
                }
              }
            >
              09123456789
            </Text>
          </View>
          <View
            style={
              Object {
                "alignItems": "baseline",
                "backgroundColor": "transparent",
                "flexDirection": "row-reverse",
                "justifyContent": "space-between",
              }
            }
          >
            <Text />
            <Text
              style={
                Object {
                  "color": "green",
                  "fontSize": 20,
                }
              }
            >
               آنلاین
            </Text>
          </View>
          <View>
            <View
              pointerEvents="auto"
              style={
                Object {
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessibilityRole="button"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessibilityRole="button"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessibilityRole="button"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessibilityRole="button"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
              <View
                forwardedRef={[Function]}
                style={Object {}}
              >
                <View
                  accessibilityRole="button"
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                >
                  <Text />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#b31414",
            "borderRadius": 10,
            "elevation": 5,
            "fontSize": 25,
            "fontWeight": "bold",
            "height": 40,
            "marginLeft": 15,
            "marginTop": 15,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.26,
            "width": 370,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "marginTop": 5,
              "textAlign": "center",
            }
          }
        >
          محصولات فروشگاه
        </Text>
      </View>
      <Text
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      />
    </View>
  </View>
</RCTScrollView>
`;
